### PR TITLE
Add interface and conditional build for fileshare process

### PR DESCRIPTION
**Pull request by bartoszWojciechO**

### CODE REVIEW

File: cmd/cli/fileshare_process.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+//go:build fileshare_process
```
Comment (bartoszWojciechO) [inline cmd/cli/fileshare_process.go:1]: GRPC fileshare process is implemented in the next commit. For now this file is not used anywhere thanks to conditional compilation.

**Review by Savolro — APPROVED** [review]

**Review by keliramu — APPROVED** [review]

File: cmd/cli/fileshare_process.go (hunk 1 above)
Comment (mariusSincovici) [reply cmd/cli/fileshare_process.go:1]: will `fileshare_process` be part of the `.env` file, added to `FEATURES`?

File: cmd/fileshare_process/main.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	// #nosec path is constant
+	logFile, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0600)
+	if err != nil {
+		return nil, err
+	}
+	return logFile, nil
+}
+
+func main() {
+	if logFile, err := openLogFile(filepath.Join(fileshare_process.FileshareLogPath, "/nordfileshared.log")); err == nil {
```
Comment (mariusSincovici) [inline cmd/fileshare_process/main.go:31]: is there any file rotation in place or something similar to restrict log file at a certain size?

File: cmd/fileshare_process/main.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		log.SetOutput(logFile)
+		log.SetFlags(log.LstdFlags | log.Lshortfile | log.Lmicroseconds)
+	}
+
+	processStatus := fileshare_process.GRPCFileshareProcess{}.ProcessStatus()
```
Comment (mariusSincovici) [inline cmd/fileshare_process/main.go:36]: [minor] maybe use the helper function `NewGRPCFileshareProcess`?

File: cmd/fileshare_process/main.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	if processStatus == fileshare_process.Running {
+		os.Exit(int(fileshare_process.CodeAlreadyRunning))
+	} else if processStatus == fileshare_process.RunningForOtherUser {
+		os.Exit(int(fileshare_process.CodeAlreadyRunningForOtherUser))
+		log.Println("Cannot start fileshare daemon, it is already running for another user.")
```
Comment (mariusSincovici) [inline cmd/fileshare_process/main.go:41]: Is message getting logged?

File: fileshare_process/grpc_fileshare_process.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+type StartupErrorCode int
+
+const (
```
Comment (mariusSincovici) [inline fileshare_process/grpc_fileshare_process.go:20]: Will this errors need to be handled also by the GUI app?

File: cmd/cli/fileshare_process.go (hunk 1 above)
Comment (bartoszWojciechO) [reply cmd/cli/fileshare_process.go:1]: As is, it's configured in `snapcraft/snap.yml. No reason to add it to `FEATURES` I think.

File: cmd/fileshare_process/main.go (hunk 4 above)
Comment (bartoszWojciechO) [reply cmd/fileshare_process/main.go:41]: I think so, logger is initialized before.

File: fileshare_process/grpc_fileshare_process.go (hunk 5 above)
Comment (bartoszWojciechO) [reply fileshare_process/grpc_fileshare_process.go:20]: Some will probably need to be, at least to display correct error messages to the user. Depends on the GUI implementation.

File: cmd/fileshare_process/main.go (hunk 2 above)
Comment (bartoszWojciechO) [reply cmd/fileshare_process/main.go:31]: We don't do rotation for local user logs afaik, probably should implement that.

File: cmd/fileshare_process/main.go (hunk 4 above)
Comment (mariusSincovici) [reply cmd/fileshare_process/main.go:41]: but there is an `os.Exit` above it. isn't that stopping immediately the process?

**Review by mariusSincovici — APPROVED** [review]

**Pull request merged**